File: 백준/레벨Silver III/11659. 구간 합 구하기 4/구간 합 구하기 4.ts
// @ts-ignore
import fs = require('fs');

const read = fs.readFileSync(0, 'utf-8');
const lines = read.split('\n').map((line: string) => line.split(' ').map(Number));

const rangeSum = () => {
  // 누적 합 초기화
  let [N, M] = lines[0];
  const prefixSum = Array(N + 1).fill(0);

  for (let i = 0; i < N; i++) {
    prefixSum[i + 1] = prefixSum[i] + lines[1][i];
  }

  // 구간 합 출력
  const result = [];
  for (let i = 0; i < M; i++) {
    const [s, e] = lines[i + 2];
    result.push(prefixSum[e] - prefixSum[s - 1]);
  }
  console.log(result.join('\n'));
};

rangeSum();

File: 백준/레벨Silver III/11659. 구간 합 구하기 4/구간 합 구하기 4.java
import java.io.*;

class Solution {
    public void rangeSum() throws IOException {
        // 누적 합 초기화
        int N = nextInt(), M = nextInt();
        int[] prefixSum = new int[N+1];

        for (int i = 0; i < N; i++) {
            prefixSum[i + 1] = prefixSum[i] + nextInt();
        }

        // 구간 합 출력
        StringBuilder sb = new StringBuilder();
        while (M-- > 0) {
            int i = nextInt(), j = nextInt();
            sb.append(prefixSum[j] - prefixSum[i - 1]).append("\n");
        }

        System.out.println(sb);
    }

    int nextInt() throws IOException {
        int n = 0;
        int c;

        while ((c = System.in.read()) <= 32); // 탭 2, 개행 10, 공백 32

        do {
            n = n * 10 + (c - '0'); // 한자리씩 추가
        } while ((c = System.in.read()) > 32); // 구분자 : 공백

        return n;
    }
}

public class Main {
    public static void main(String[] args) throws IOException {
        new Solution().rangeSum();
    }
}